File: src/output.rs
pub enum OutputKind {
	Info,
	Warn,
	Error,
}

impl std::fmt::Display for OutputKind {
	fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
		match self {
			OutputKind::Info => write!(f, "info"),
			OutputKind::Warn => write!(f, "warn"),
			OutputKind::Error => write!(f, "error"),
		}
	}
}

pub trait Output {
	fn init(&mut self) {}
	fn write(&mut self, kind: OutputKind, message: String);
	fn report(&mut self, file: &str, message: &str, got: &str, expected: &str);
	fn end(&mut self) {}
}

File: src/output/Output.cpp
//
// Created by ENDERZOMBI102 on 15/10/2023.
//
#include "Output.hpp"


auto toString( const OutputKind kind ) -> std::string_view {
	switch ( kind ) {
		case OutputKind::Info: return "info";
		case OutputKind::Warn: return "warn";
		case OutputKind::Error: return "error";
	}
	return {};
}
